ptypes/struct.go

```diff
@@ -36,6 +39,23 @@ type Struct struct {
 	types.Struct
 }
 
+func (s *Struct) EncodeMsgpack(encoder *msgpack.Encoder) error {
+	raw, err := s.MarshalJSONPB(defaultJSONPBMarshaler)
```

Comment:
Isn't it better to use `proto.Marsha` instead?

Reply:
Good idea.